### PR TITLE
Separate stderr from stdout

### PR DESCRIPTION
**Pull request by erikeah**
This PR solves #17 .

Excuse me by late PR, I hope you are still interested in this feature.

### CODE REVIEW

**Comment by jonaslu** [comment]
Sure and thanks, it still makes sense. I'll have a look in the coming days.

File: internal/pkg/call/call.go [hunk 1]
```diff
@@ -64,22 +71,31 @@ func ValidBackend(backendName string) bool {
...
-		return "", errors.Wrapf(err, "Could not instantiate backend: %s", callData.Backend)
+		return Output{}, errors.Wrapf(err, "Could not instantiate backend: %s", callData.Backend)
 	}
 
 	if printCommand {
-		return backend.getAsString()
+		if command, err := backend.getAsString(); err != nil {
+			return Output{StdErr: []byte(command)}, err
+		} else {
+			return Output{StdOut: []byte(command)}, nil
```
Comment (jonaslu) [inline internal/pkg/call/call.go:89]: Change getAsString() into returning a []byte array with a matching method name, it's currently converting a string to a []byte and then into a string again

File: internal/pkg/call/call.go [hunk 2]
```diff
@@ -11,6 +13,11 @@ import (
 	"github.com/pkg/errors"
 )
 
+type Output struct {
```
Comment (jonaslu) [inline internal/pkg/call/call.go:16]: Output is too generic, I'd call it BackendOutput

**Review by jonaslu — CHANGES_REQUESTED** [review]
Test plan: None is only ok if it's a textual (non functional change such as pure refactor or changes in say a README.md file). Also please keep the test-plan formatting. It'll be easier on the eyes. I'll submit a .gitmessage template.

Any functional change needs tests that others can run and reproduce.
It helps to think in terms of code-coverage: make sure you've covered all branches of the code you've changed (especially cases where errors are returned).

**Comment by erikeah** [comment]
I will give a try once I've free time, thanks for feedback, makes sense!

**Comment by jonaslu** [comment]
This has been fixed in eb9fa6c, closing this

**Pull request closed without merging**